### PR TITLE
config: hide & deprecate enable-streaming

### PR DESCRIPTION
**Pull request by SunRunAway**
### What is changed, added or deleted? (Required)

`enable-streaming` is not well maintained because we are focusing on optimizing the non-streaming coprocessor API for its performance and resource usage and don’t have time to optimize streaming's currently, I propose to deprecate it temporarily until somebody has time to pick it up again and prove it could be GA in the future.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/20760

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Have version specific changes 
- [x] Might cause conflicts

### CODE REVIEW

**Review by breeswish — APPROVED** [review]

**Comment by TomShawn** [comment]
release-3.1 分支不需要去掉这个变量吗？

**Comment by breeswish** [comment]
需要的，请加一下，谢谢！

**Comment by TomShawn** [comment]
> 需要去掉，请加一下，谢谢！

文档和代码保持一致。如果 v3.1 的代码中去掉了，文档中也去掉。反之亦然。

**Comment by breeswish** [comment]
@TomShawn 3.1 版本代码已经不维护了，不会发布新版本..

**Comment by TomShawn** [comment]
> @TomShawn 3.1 版本代码已经不维护了，不会发布新版本..

那就不 pick 到 3.1 了。保持一致就好。

**Review by TomShawn — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-show-variables.md [hunk 1]
```diff
@@ -69,7 +69,6 @@ SHOW GLOBAL VARIABLES LIKE 'tidb%';
 | tidb_enable_radix_join              | 0                   |
 | tidb_enable_slow_log                | 1                   |
 | tidb_enable_stmt_summary            | 1                   |
-| tidb_enable_streaming               | 0                   |
 | tidb_enable_table_partition         | on                  |
```
Comment (TomShawn) [inline sql-statements/sql-statement-show-variables.md:72]:
```suggestion
| tidb_enable_table_partition          | on                  |
```

**Pull request merged**